arcgis-ios-sdk-samples/Layers/Display KML network links/DisplayKMLNetworkLinksViewController.swift

```diff
@@ -57,4 +62,34 @@ class DisplayKMLNetworkLinksViewController: UIViewController {
         scene.operationalLayers.add(kmlLayer)
     }
     
+    private func updatePopoverForMessage() {
+        // updates the text shown in the text view with the message text
+        messageViewController?.textView?.text = messageText
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // The popovover segue is triggered via storyboard connections, so set it up here.
+        if let controller = segue.destination as? KMLNetworkMessagesViewController {
+            controller.presentationController?.delegate = self
+            controller.preferredContentSize = CGSize(width: 300, height: 300)
+            // load the view so the text view can be accessed now
+            controller.loadViewIfNeeded()
+            // retain a reference to the controller so the text view can be updated from `updatePopoverForMessage()`
+            messageViewController = controller
+            // set the initial message
+            updatePopoverForMessage()
+        }
+    }
+}
+
+extension DisplayKMLNetworkLinksViewController: UIAdaptivePresentationControllerDelegate {
+    
+    func adaptivePresentationStyle(for controller: UIPresentationController, traitCollection: UITraitCollection) -> UIModalPresentationStyle {
+        // show as this as a popover even on small displays
+        return .none
+    }
+}
+
+class KMLNetworkMessagesViewController: UIViewController {
```

Comment:
A view controller without any custom logic is just a view. A view controller mediates between a model and a view. What's the model for this view controller? It should be the message. The consumer of `KMLNetworkMessagesViewController` should be able to assign a message to the view controller, whether its view has loaded or not, and the view controller itself should update its own view appropriately.